components/server/ee/src/user/user-counter.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { injectable } from 'inversify';
+
+@injectable()
+export class UserCounter {
+    public expires: Date | null = null;
+
+    protected data: number | null = null;
+
+    protected readonly timeout: number = 60 * 1000; // Cache data for 1 minute
+
+    get count(): number | null {
```

Comment:
nit: we usually do not use `null` but `undefined` instead. But ignore this for how, we'll have a lint for it shortly.

Reply:
@geropl Would you like me to change this? I don't mind - normally, I use `undefined` as well so I can't understand why I used `null`